YogaKit/Source/YGLayout.m

```diff
@@ -451,15 +451,26 @@ static void YGApplyLayoutToViewHierarchy(UIView *view, BOOL preserveOrigin)
     topLeft.y + YGNodeLayoutGetHeight(node),
   };
 
-  const CGPoint origin = preserveOrigin ? view.frame.origin : CGPointZero;
-  view.frame = (CGRect) {
-    .origin = {
-      .x = YGRoundPixelValue(topLeft.x + origin.x),
-      .y = YGRoundPixelValue(topLeft.y + origin.y),
-    },
+  const CGPoint origin = preserveOrigin ? (CGPoint){
+    .x = view.center.x - (view.bounds.size.width / 2),
+    .y = view.center.y - (view.bounds.size.height / 2),
+  } : CGPointZero;
+
+  CGFloat x = topLeft.x + origin.x;
+  CGFloat y = topLeft.y + origin.y;
+  CGFloat width = bottomRight.x - topLeft.x;
+  CGFloat height = bottomRight.y - topLeft.y;
+
+  view.center = (CGPoint) {
```

Comment:
Hi @lelandrichardson, to handle correctly the view's layer.anchorPoint and to keep the view's bounds.origin, you should update your code to something like that. This thing is little trickyé UIScrollView's bounds.origin is not .zero when a contentOffset is set.

``` 
// NOTE: The center is offset by the layer.anchorPoint, so we have to take it into account.
    view.center = (CGPoint) {
        .x = YGRoundPixelValue(x + (width * view.layer.anchorPoint.x)),
        .y = YGRoundPixelValue(y + (height * view.layer.anchorPoint.y)),
    };

    // NOTE: We must set only the bounds' size and keep the origin.
    view.bounds = (CGRect) {
        .origin = view.bounds.origin,
        .size = {
            .width = YGRoundPixelValue(width),
            .height = YGRoundPixelValue(height),
        },
    };
```